src/transformers/utils/fx.py

```diff
@@ -22,16 +24,93 @@
+from ..models.electra import ElectraConfig
+from ..models.gpt2 import GPT2Config
+from ..models.gpt_neo import GPTNeoConfig
+from ..models.gptj import GPTJConfig
+from ..models.megatron_bert import MegatronBertConfig
+from ..models.mobilebert import MobileBertConfig
+from ..models.t5 import T5Config
 
 
 logger = logging.get_logger(__name__)
 
 
+T = TypeVar("T")
+TypeOrListOfType = Union[T, List[T]]
+
+
+def _generate_supported_model_classes(
+    model_config_class: Type[PretrainedConfig],
+    supported_tasks: Optional[TypeOrListOfType[str]] = None,
+    other_model_classes: Optional[TypeOrListOfType[Type[PretrainedConfig]]] = None,
+):
+    task_mapping = {
+        "default": MODEL_MAPPING,
+        "pretraining": MODEL_FOR_PRETRAINING_MAPPING,
+        "next-sentence-prediction": MODEL_FOR_NEXT_SENTENCE_PREDICTION_MAPPING,
+        "masked-lm": MODEL_FOR_MASKED_LM_MAPPING,
+        "causal-lm": MODEL_FOR_CAUSAL_LM_MAPPING,
+        "seq2seq-lm": MODEL_FOR_SEQ_TO_SEQ_CAUSAL_LM_MAPPING,
+        "multiple-choice": MODEL_FOR_MULTIPLE_CHOICE_MAPPING,
+        "question-answering": MODEL_FOR_QUESTION_ANSWERING_MAPPING,
+        "sequence-classification": MODEL_FOR_SEQUENCE_CLASSIFICATION_MAPPING,
+        "token-classification": MODEL_FOR_TOKEN_CLASSIFICATION_MAPPING,
+        "image-classification": MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING,
+    }
+
+    if supported_tasks is None:
+        supported_tasks = task_mapping.keys()
+    if isinstance(supported_tasks, str):
+        supported_tasks = [supported_tasks]
+
+    model_classes = []
+    for task in supported_tasks:
+        model_class = task_mapping[task].get(model_config_class, None)
+        if model_class:
+            model_classes.append(model_class)
+
+    if other_model_classes:
+        if not isinstance(other_model_classes, (tuple, list)):
+            other_model_classes = [other_model_classes]
+        model_classes.extend(other_model_classes)
+
+    return tuple(model_classes)
+
+
+_SUPPORTED_MODELS = (
+    *_generate_supported_model_classes(AlbertConfig),
+    *_generate_supported_model_classes(BertConfig),
+    *_generate_supported_model_classes(DistilBertConfig),
+    *_generate_supported_model_classes(MobileBertConfig),
+    *_generate_supported_model_classes(ElectraConfig),
+    *_generate_supported_model_classes(MegatronBertConfig),
+    *_generate_supported_model_classes(GPT2Config, other_model_classes=GPT2DoubleHeadsModel),
+    *_generate_supported_model_classes(GPTJConfig),
+    *_generate_supported_model_classes(GPTNeoConfig),
+    *_generate_supported_model_classes(T5Config),
+)
+
+_SUPPORTED_MODELS_FOR_DYNAMIC_AXES = (
+    *_generate_supported_model_classes(AlbertConfig),
+    *_generate_supported_model_classes(BertConfig),
+    *_generate_supported_model_classes(DistilBertConfig),
+    *_generate_supported_model_classes(MobileBertConfig),
+    *_generate_supported_model_classes(ElectraConfig),
+    *_generate_supported_model_classes(MegatronBertConfig),
+)
```

Comment:
This would be better if the `_generate_supported_model_classes` took a model type ("albert", "bert" etc) instead of a live object. Also maybe put all supported model types in a list and all special models (like `GPT2DoubleHeadsModel`) in another list, then build those?